cfme/tests/containers/test_ssl_providers.py

```diff
@@ -148,3 +150,29 @@ def test_add_mertics_provider_ssl(provider, appliance, test_item,
     else:
         new_provider.delete(cancel=False)
         new_provider.wait_for_delete()
+
+
+@pytest.mark.parametrize('sec_protocol', AVAILABLE_SEC_PROTOCOLS)
+def test_setup_with_wrong_port(provider, sec_protocol, sync_ssl_certificate):
+    """
+    Negative test: set a provider with wrong api port
+    based on BZ1443520
+    """
+    new_provider = copy(provider)
+    new_provider.endpoints["default"].api_port = "1234"
+    new_provider.endpoints["default"].sec_protocol = sec_protocol
+
+    is_fail = False
+    try:
+        new_provider.setup()
+    except AssertionError as ex:
+        node = new_provider.endpoints["default"].hostname
+        port = new_provider.endpoints["default"].api_port
+        assert (
+            'Credential validation was not successful: Failed to open TCP connection'
+            ' to {node}:{port} (No route to host - connect(2) for "{node}" port '
+            '{port}'.format(node=node, port=port) in ex.message), (
+            "No clear failure massage was provided for wrong api port")
+        is_fail = True
+
+    assert is_fail, "Provider was set with wrong api port"
```

Comment:
**REQUIRED**
Please use `pytest.raises` context manager:
```python
with pytest.raises(SomeExceptionHere):
    new_provider.setup()
```
```python

Reply:
@quarckster Could you please explain what is the advantage of this code over my suggestion?